rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/Costs.scala

```diff
@@ -141,6 +141,9 @@ trait Costs {
   // if underlying string is shorter then the `to` value.
   def sliceCost(to: Int): Cost = Cost(to, "slice")
 
+  def splitCost(str: String, sep: String, found: Int): Cost =
+    Cost(str.getBytes.size + found * sep.length, "split")
```

Comment:
Why cost defined such way? Is cost equals to used memory? If it is, part `found * sep.length` is not transparent for me :thinking: I thought, cost is based on computational complexity.

Reply:
Computation cost should also include cost of used memory.

@hilltracer also made a response for charging calculation on the very first copy of this PR https://github.com/rchain/rchain/pull/3753#discussion_r910223400.

Reply:
It is very naive assumption. Without knowing how real algorithm is implemented we cannot accurate estimate cost. One approach can be to measure performance with different strings and separators and build some e.g. linear model. But perhaps it is such complicated way that we want to avoid.

Reply:
I agree with you, we cannot be completely accurate. It would be reasonable to understand worst case scenario and charge in that way.

If you have idea how we can create some simple model to judge I'd be very much in favor to do that. This can also help us later in the implementation of other string methods.

Reply:
A cost model for the simple split would be perfect.
There are regular expression subsets that can be implemented in finite state machines (giving us simple regex support), which could build on such a model for cost accounting.

Reply:
> If you have idea how we can create some simple model to judge I'd be very much in favor to do that. This can also help us later in the implementation of other string methods.

We can create simple code wrapper which will form table of inputs / measured output. E.g. in this case inputs will be source string length and separator length. And output will be measured time. Then simple model like
`cost = a * str_len + b * sep_len + c`
can be easily created with Python or even Excel.

![image](https://user-images.githubusercontent.com/1443855/185695667-e08d0ed4-679d-43ff-abe2-469d92e07eaf.png)

But problem is that we cannot use model's output as absolute value directly. Only as relative value to compare costs between themselves.

Reply:
@stanislavlyalin What is the meaning if the negative values on your graph?

We cannot calculate cost by the time of execution so if the measurement shows that CPU and memory use is linear to the input values current definition of the cost makes sense.
We cannot compare cost to some absolute representation for CPU and memory use, but only to compare to other cost calculation and be similar in a relative sense.

---

> @Bill-Kunj There are regular expression subsets that can be implemented in finite state machines (giving us simple regex support), which could build on such a model and give us regex cost accounting.

Without analyzing regex expressions we cannot know which expressions can cause exponential use of CPU and memory so calculation for regex is not feasible.
Some examples where regex can cause unexpected CPU usage.
https://stackoverflow.com/questions/17386784/regex-causing-abnormally-high-cpu-use-in-c-sharp
https://stackoverflow.com/questions/6230489/100-cpu-usage-with-a-regexp-depending-on-input-length

Reply:
This graph is just linear model example. NVM. For non-linear relations we can use little bit more complex models like a `cost = ax^2 + bx + c`. Usually we can look at the measurements and decide which model is more appropriate in concrete case. And we can predict real cost of operation if we choose some deterministic operation as base, e.g. summation.